apps/cowswap-frontend/src/legacy/state/orders/utils.ts

```diff
@@ -425,6 +430,21 @@ export function getRemainderAmountsWithoutSurplus(order: Order | ParsedOrder): {
  */
 export function getRemainderAmount(kind: OrderKind, order: Order | ParsedOrder): string {
   const buyAmount = order.buyAmount.toString()
+
+  const { sellAmount, executedSellAmount, executedBuyAmount } = getExecutedAmounts(order)
+
+  const fullAmount = isSellOrder(kind) ? sellAmount : buyAmount
+  
+  if (!executedSellAmount || !executedBuyAmount || executedSellAmount === '0' || executedBuyAmount === '0') {
+    return fullAmount
+  }
+
+  const executedAmount = JSBI.BigInt((isSellOrder(kind) ? executedSellAmount : executedBuyAmount) || 0)
+
+  return JSBI.subtract(JSBI.BigInt(Number(fullAmount)), executedAmount).toString()
```

Comment:
Here we also should use .toString(16) instead of casting to Number in order to not loose precision